normandy/base/api/authentication.py

```diff
@@ -123,3 +123,45 @@ def fetch_oidc_user_profile(self, access_token):
         # This could happen if, for some reason, we're not configured to be
         # allowed to talk to the OIDC endpoint.
         raise OIDCEndpointRequestError(response.status_code)
+
+
+class InsecureEmailAuthentication(BaseAuthentication):
+    """
+    A simple but insecure way to authenticate users for QA/dev purposes.
+
+    Clients should authenticate by passing the token key in the "Authorization" HTTP header,
+    prepended with the string "Bearer ".  For example:
+
+        Authorization: Insecure jdoe@mozilla.com
+    """
+
+    keyword = "Insecure"
+
+    def authenticate(self, request):
+        auth_header = get_authorization_header(request).decode().split()
+
+        if not auth_header or auth_header[0].lower() != self.keyword.lower():
+            return None
+
+        email = auth_header[1]
+
+        return self.authenticate_credentials(email)
+
+    def authenticate_credentials(self, email):
+        # Turn this email into a Django User instance.
+        user, _ = get_user_model().objects.get_or_create(
```

Comment:
That's definitely an approach too. I guess it really depends on what we're trying to achieve from a testing perspective.

Reply:
@jrbenny35 do you have an opinion here? Do you think that local Normandy/Delivery Console authentication should automatically create new users? Or would you rather make users manually and have typo-protection when you switch users?

Reply:
For my use case, creating the users manually would be best, as I would be able to specifically check for that user if there are test cases revolving around a user (which I think there are). There could be cases for having a user automatically created for someone just wanting to poke around with a local instance. Don't really know how much that would get used though.

Reply:
I mean another option is to support both with an additional keyword

Reply:
Maybe something like:
- `Insecure Create {token}` to create a new user
- `Insecure Login {token}` to log in only

Reply:
But if the create case is unlikely to get used I'm happy to scrap it altogether

Reply:
It's easy enough to create users via the API, so I think that the `Insecure create` case is niche enough we don't need to use it. It seems like it would be easy enough to add in the future if that's not right.